ProtocolHandler.pm

```diff
@@ -268,11 +270,26 @@ sub getMetadataFor {
 	}
 
 	$meta->{title} = Plugins::Qobuz::API::Common->addVersionToTitle($meta);
-	
-	if ($meta->{work}) {
-		$meta->{title} = $meta->{work} . ': ' . $meta->{title};
-	}
 
+	# user pref is for enhanced classical music display, and we have a classical release	
+	if ( $meta->{isClassique} ) {
+		# if the title doesn't already contain the work text
+		if ( $meta->{work} && index($meta->{title},$meta->{work}) == -1 ) {
+			$meta->{title} =~ s/${\(split " ", $meta->{composer})[-1]}:\s*// if $meta->{composer};
```

Comment:
Hehe... I think that's one of those lines which gave Perl a bad reputation 😂. It's like four or five instructions in a single line, seasoned with some regex magic. Please move the split out of the regex and give it a name for understandability.

And whenever you use input data as part of a regex make sure you "escape" it using `/\Q$variable/`. Otherwise you risk to crash the server if eg. the value contained a slash or other meta character used in regular expressions.

https://perldoc.perl.org/functions/quotemeta

Reply:
OK, I've replaced with the following, in which I've also enhanced to code to check for the entire composer name as well as only the surname. I've never seen the whole composer name in a track title, but you never know...

```
	if ( $meta->{isClassique} ) {
		# if the title doesn't already contain the work text
		if ( $meta->{work} && index($meta->{title},$meta->{work}) == -1 ) {
			# remove composer name from track title
			if ( $meta->{composer} ) {
				# full name
				$meta->{title} =~ s/\Q$meta->{composer}\E:\s*//;
				# surname only
				my $composerSurname = (split " ", $meta->{composer})[-1];
				$meta->{title} =~ s/\Q$composerSurname\E:\s*//;
			}

```
Does this look OK?

Reply:
That does add removal of the composer's full name? But yes, looking much better - thanks!